pkg/drivers/jetstream/jetstream.go

```diff
@@ -0,0 +1,1017 @@
+package jetstream
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/k3s-io/kine/pkg/drivers/jetstream/kv"
+	"github.com/k3s-io/kine/pkg/server"
+	"github.com/k3s-io/kine/pkg/tls"
+	"github.com/nats-io/jsm.go/natscontext"
+	"github.com/nats-io/nats.go"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	defaultBucket       = "kine"
+	defaultRevHistory   = 10
+	defaultSlowMethodMs = 500
+)
+
+var (
+	toplevelKeyMatch = regexp.MustCompile(`(/[^/]*/[^/]*)(/.*)?`)
+)
+
+type Config struct {
+	natsURL          string
+	options          []nats.Option
+	revHistory       uint8
+	bucket           string
+	slowMilliseconds int64
+}
+
+type JetStream struct {
+	kvBucket         nats.KeyValue
+	kvBucketMutex    *sync.RWMutex
+	kvDirectoryMutex *sync.RWMutex
+	kvDirectoryMuxes map[string]*sync.RWMutex
+	jetStream        nats.JetStreamContext
+	slowMethod       int64
```

Comment:
A more idiomatic way to specify this would be to use a `time.Duration` instead of an integer number of milliseconds; this would save you a bit of code on the parameter parsing side too.

Reply:
I thought about doing that but then thought maybe that would provide too much flexibility... i.e. someone says a slow method is `1hr` but using a duration would probably make more sense